packages/client/wallets/smart-wallet/src/SmartWalletSDK.ts

```diff
@@ -48,7 +60,14 @@ export class SmartWalletSDK extends LoggerWrapper {
         return logPerformance(
             "GET_OR_CREATE_WALLET",
             async () => {
-                return await this.smartWalletService.getOrCreate(user, chain, walletConfig);
+                try {
```

Comment:
It might be cool to use a `Proxy` for this pattern, we do this in a bunch of places in the SDK

Reply:
There's 2 instances atm I think, I wanna wait a little longer until there's 3 or 4 before addressing, will be able to make a better abstraction with more cases.